typescript/src/main.ts

```diff
@@ -0,0 +1,298 @@
+        const response = await this.httpPutDigitalFactory(
+            `${API_ROOT_URL}/cura/v1/projects/${projectId}/comments`,
+            {
+                data: {
+                    body: comment
+                }
+            }
+        );
+        return response.data;
+    }
+
+    async submitPrintJob(jobId: string): Promise<any> {
+        const clusterId = process.env.CLUSTER_ID
+        const response = await this.httpPostDigitalFactory(
+            `${API_ROOT_URL}/connect/v1/clusters/${clusterId}/print/${jobId}`,
+            { data: {}}
+        );
+        return response.data;
+    }
+
+    async getRunningPrintJobs(): Promise<any> {
+        const query = new URLSearchParams({
+            limit: "20",
+            status: "in_progress"
+        });
+        const response = await this.httpGetDigitalFactory(`${API_ROOT_URL}/connect/v1/print_jobs?${query}`);
+        return response.data;
+    }
+
+    async searchProjects(): Promise<any> {
+        const query = new URLSearchParams({
+            limit: "24",
+            page: "1",
+            search: "demo",
+            shared: "false"
+        });
+        const response = await this.httpGetDigitalFactory(`${API_ROOT_URL}/cura/v1/projects?${query}`);
+        return response.data;
+    }
+}
+
+async function main(): Promise<void> {
+    const demo = new DigitalFactoryDemo();
+    await demo.signIn();
+    print("Sign in completed.\n");
+
+    print("Creating demo project...");
+    const { library_project_id } = await demo.createProject("Demo project");
+    print(`Created project with ID: ${library_project_id}\n`);
+
+    print("Adding comment to demo project...");
+    await demo.addCommentToProject(library_project_id, "Demo comment");
+    print("Comment added.\n");
+
+    print("Uploading file to demo project...");
+    const { job_id } = await demo.uploadFileToProject(library_project_id);
+    print(`Uploaded file with ID: ${job_id}\n`);
+    print(`Visit https://digitalfactory.ultimaker.com/app/library/project/${library_project_id} to see your project\n`);
+
+    print("Submitting a print job");
+    const { job_instance_uuid } = await demo.submitPrintJob(job_id);
+    print(`Submitted print job with ID: ${job_instance_uuid}\n`);
+
+    print("Getting running print jobs.");
+    const printJobs = await demo.getRunningPrintJobs();
+    if (printJobs.length > 0) {
+        print(`Total print jobs retrieved: ${printJobs.length}`);
+        print(`First print job retrieved: ${JSON.stringify(printJobs[0])}\n`);
+    } else {
+        print("No running print jobs found. Sometimes it takes up to 10 second for new print jobs to show up.\n");
+    }
+
+
+    print("Searching projects.")
+    const projects = await demo.searchProjects();
+    if (projects.length > 0) {
+        print(`Total projects retrieved: ${projects.length}`);
+        print(`First project retrieved: ${JSON.stringify(projects[0])}\n`);
```

Comment:
`JSON.stringify(projects[0], null, 4)` to make the printing contents more readable?

Reply:
That does sound like a good idea.

Reply:
done